src/Core/Echo.Ast/InstructionExpression.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Echo.ControlFlow.Serialization.Dot;
+using Echo.Core.Graphing;
+
+namespace Echo.Ast
+{
+    /// <summary>
+    /// Represents an instruction in the AST
+    /// </summary>
+    public sealed class InstructionExpression<TInstruction> : Expression<TInstruction>
+    {
+        /// <summary>
```

Comment:
Add constructor for expressions without arguments, and a `params Expression<TInstruction>[]` for consistency and readability.